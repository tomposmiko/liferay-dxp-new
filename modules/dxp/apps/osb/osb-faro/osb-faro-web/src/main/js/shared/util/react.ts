import {isEqual} from 'lodash';

export const getDisplayName = WrappedComponent =>
	WrappedComponent.displayName || WrappedComponent.name || 'Component';

type HasChanges = <T>(
	prev: T | object,
	next: T | object,
	...keys: string[]
) => boolean;

/**
 * Compare previous state or props object by provided keys to detect changes.
 */
export const hasChanges: HasChanges = (prev = {}, next = {}, ...keys) => {
	for (const key of keys) {
		if (key in next) {
			const newVal = next[key];

			const prevVal = prev[key];

			if (!isEqual(newVal, prevVal)) {
				return true;
			}
		}
	}

	return false;
};
