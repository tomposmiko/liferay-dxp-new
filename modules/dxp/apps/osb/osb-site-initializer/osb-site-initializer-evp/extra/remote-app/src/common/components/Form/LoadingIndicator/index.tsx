/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * The contents of this file are subject to the terms of the Liferay Enterprise
 * Subscription License ("License"). You may not use this file except in
 * compliance with the License. You can obtain a copy of the License by
 * contacting Liferay, Inc. See the License for the specific language governing
 * permissions and limitations under the License, including but not limited to
 * distribution rights of the Software.
 */

import ClayLoadingIndicator from '@clayui/loading-indicator';

const LoadingIndicator = () => {
	return (
		<div className="align-items-center d-flex flex-wrap justify-content-center">
			<ClayLoadingIndicator
				displayType="primary"
				shape="squares"
				size="lg"
			/>
		</div>
	);
};

export default LoadingIndicator;
