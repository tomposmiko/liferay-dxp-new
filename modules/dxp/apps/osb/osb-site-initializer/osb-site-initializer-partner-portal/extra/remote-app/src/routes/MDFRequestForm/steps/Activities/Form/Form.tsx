/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * The contents of this file are subject to the terms of the Liferay Enterprise
 * Subscription License ("License"). You may not use this file except in
 * compliance with the License. You can obtain a copy of the License by
 * contacting Liferay, Inc. See the License for the specific language governing
 * permissions and limitations under the License, including but not limited to
 * distribution rights of the Software.
 */

import React, {useCallback, useMemo, useState} from 'react';

import PRMForm from '../../../../../common/components/PRMForm';
import PRMFormik from '../../../../../common/components/PRMFormik';
import {LiferayPicklistName} from '../../../../../common/enums/liferayPicklistName';
import MDFRequestActivity from '../../../../../common/interfaces/mdfRequestActivity';
import getBooleanEntries from '../../../../../common/utils/getBooleanEntries';
import BudgetBreakdownSection from './components/BudgetBreakdownSection';
import ContentMarketingFields from './components/ContentMarketingFields';
import DigitalMarketingFields from './components/DigitalMarketingFields';
import EventFields from './components/EventFields';
import MiscellaneousMarketingFields from './components/MiscellaneousMarketingFields';
import {TypeActivityExternalReferenceCode} from './enums/typeActivityExternalReferenceCode';
import useDynamicFieldEntries from './hooks/useDynamicFieldEntries';
import useSelectedTypeActivity from './hooks/useSelectedTypeActivity';
import useTacticsTypeActivity from './hooks/useTacticsTypeActivity';

interface IProps {
	currentActivity: MDFRequestActivity;
	currentIndex: number;
	setFieldValue: (
		field: string,
		value: any,
		shouldValidate?: boolean | undefined
	) => void;
}

type TypeOfActivityComponent = {
	[key in string]?: JSX.Element;
};

const Form = ({currentActivity, currentIndex, setFieldValue}: IProps) => {
	const [tacticsByTypeActivity, setTacticsByTypeActivity] = useState<
		React.OptionHTMLAttributes<HTMLOptionElement>[]
	>();

	const {fieldEntries, typeOfActivities} = useDynamicFieldEntries();

	const selectedTypeActivity = useSelectedTypeActivity(
		currentActivity,
		typeOfActivities
	);

	const typeOfActivitiesEntries = useMemo(
		() =>
			typeOfActivities?.map((typeOfActivity) => ({
				label: typeOfActivity.name,
				value: String(typeOfActivity.id),
			})),
		[typeOfActivities]
	);

	useTacticsTypeActivity(
		selectedTypeActivity,
		useCallback(
			(tactics) =>
				setTacticsByTypeActivity(
					tactics.map((tatic) => ({
						label: tatic.name,
						value: tatic.id,
					}))
				),
			[]
		)
	);

	const typeOfActivityComponents: TypeOfActivityComponent = {
		[TypeActivityExternalReferenceCode.DIGITAL_MARKETING]: (
			<DigitalMarketingFields index={currentIndex} />
		),
		[TypeActivityExternalReferenceCode.CONTENT_MARKETING]: (
			<ContentMarketingFields index={currentIndex} />
		),
		[TypeActivityExternalReferenceCode.EVENT]: (
			<EventFields index={currentIndex} />
		),
		[TypeActivityExternalReferenceCode.MISCELLANEOUS_MARKETING]: (
			<MiscellaneousMarketingFields index={currentIndex} />
		),
	};

	return (
		<>
			<PRMForm.Section title="Campaign Activity">
				<PRMFormik.Field
					component={PRMForm.InputText}
					label="Activity name"
					name={`activities[${currentIndex}].name`}
					required
				/>

				<PRMForm.Group>
					<PRMFormik.Field
						component={PRMForm.Select}
						label="Type of Activity"
						name={`activities[${currentIndex}].r_typeActivityToActivities_c_typeActivityId`}
						options={typeOfActivitiesEntries}
						required
					/>

					<PRMFormik.Field
						component={PRMForm.Select}
						emptyOptionMessage="Select a Type of Activity"
						label="Tactic"
						name={`activities[${currentIndex}].r_tacticToActivities_c_tacticId`}
						options={tacticsByTypeActivity}
						required
					/>
				</PRMForm.Group>

				{
					typeOfActivityComponents[
						selectedTypeActivity?.externalReferenceCode || ''
					]
				}

				<PRMForm.Group>
					<PRMFormik.Field
						component={PRMForm.DatePicker}
						label="Start Date"
						name={`activities[${currentIndex}].startDate`}
						required
					/>

					<PRMFormik.Field
						component={PRMForm.DatePicker}
						label="End Date"
						name={`activities[${currentIndex}].endDate`}
						required
					/>
				</PRMForm.Group>
			</PRMForm.Section>
			<PRMFormik.Array
				budgets={currentActivity?.budgets}
				component={BudgetBreakdownSection}
				currentActivityIndex={currentIndex}
				expenseEntries={
					fieldEntries[LiferayPicklistName.BUDGET_EXPENSES]
				}
				name={`activities[${currentIndex}].budgets`}
				setFieldValue={setFieldValue}
			/>
			<PRMForm.Section title="Lead List">
				<PRMFormik.Field
					component={PRMForm.RadioGroup}
					items={getBooleanEntries()}
					label="Is a lead list an outcome of this activity?"
					name={`activities[${currentIndex}].leadGenerated`}
					required
					small
				/>

				<PRMFormik.Field
					component={PRMForm.InputText}
					label="Target # of Leads"
					name={`activities[${currentIndex}].targetofLeads`}
					required
				/>

				<PRMFormik.Field
					component={PRMForm.CheckboxGroup}
					items={
						fieldEntries[
							LiferayPicklistName.LEAD_FOLLOW_UP_STRATEGIES
						]
					}
					label="Lead Follow Up strategy (select all that apply)"
					name={`activities[${currentIndex}].leadFollowUpStrategies`}
					required
				/>

				<PRMFormik.Field
					component={PRMForm.InputText}
					description="(i) Please describe the follow-up plan in detail:  Do you need any assets from Liferay (i.e.  landing page, collateral, content) Will Liferay participate in the follow up? If so, please provide details"
					label="Details on Lead Follow Up. What to include (i)"
					name={`activities[${currentIndex}].detailsLeadFollowUp`}
					required
				/>
			</PRMForm.Section>
		</>
	);
};

export default Form;
