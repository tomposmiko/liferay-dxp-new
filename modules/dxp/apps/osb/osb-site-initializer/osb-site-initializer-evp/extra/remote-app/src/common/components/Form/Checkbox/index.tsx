/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * The contents of this file are subject to the terms of the Liferay Enterprise
 * Subscription License ("License"). You may not use this file except in
 * compliance with the License. You can obtain a copy of the License by
 * contacting Liferay, Inc. See the License for the specific language governing
 * permissions and limitations under the License, including but not limited to
 * distribution rights of the Software.
 */

import {ClayCheckbox} from '@clayui/form';
import {useState} from 'react';

type CheckboxProps = {
	inline?: boolean;
	label?: string;
} & React.InputHTMLAttributes<HTMLInputElement>;

const Checkbox: React.FC<CheckboxProps> = (props) => {
	const [value, setValue] = useState(false);

	return (
		<ClayCheckbox
			checked={value}
			onChange={() => setValue(!value)}
			{...props}
		/>
	);
};

export default Checkbox;
