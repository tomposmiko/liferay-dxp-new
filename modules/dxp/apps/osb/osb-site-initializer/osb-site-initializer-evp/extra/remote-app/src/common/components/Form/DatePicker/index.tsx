/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * The contents of this file are subject to the terms of the Liferay Enterprise
 * Subscription License ("License"). You may not use this file except in
 * compliance with the License. You can obtain a copy of the License by
 * contacting Liferay, Inc. See the License for the specific language governing
 * permissions and limitations under the License, including but not limited to
 * distribution rights of the Software.
 */

import ClayDatePicker from '@clayui/date-picker';
import dayjs from 'dayjs';
import {InputHTMLAttributes, useEffect, useState} from 'react';
import {UseFormRegister} from 'react-hook-form';

import {generateReportsType} from '../../../../routes/reports/generateReport';
import BaseWrapper from '../Base/BaseWrapper';

type DatePickerTypes = {
	clearErrors?: any;
	errors?: any;
	id?: string;
	label?: string;
	name: string;
	register?: UseFormRegister<generateReportsType>;
	required?: boolean;
	setValue?: any;
	type?: string;
	value: string;
} & InputHTMLAttributes<HTMLInputElement>;

const DatePicker: React.FC<DatePickerTypes> = ({
	clearErrors = {},
	errors = {},
	label,
	name,
	id = name,
	required = false,
	setValue = () => {},
	value,
}) => {
	const [data, setData] = useState(value);

	useEffect(() => {
		clearErrors(name);
		setValue(name, data);
	}, [clearErrors, data, name, setValue]);

	return (
		<BaseWrapper
			error={errors[name]?.message}
			id={id}
			label={label}
			required={required}
		>
			<ClayDatePicker
				onChange={setData}
				placeholder="YYYY-MM-DD"
				value={data}
				years={{
					end: dayjs().year(),
					start: 1997,
				}}
			/>
		</BaseWrapper>
	);
};

export default DatePicker;
