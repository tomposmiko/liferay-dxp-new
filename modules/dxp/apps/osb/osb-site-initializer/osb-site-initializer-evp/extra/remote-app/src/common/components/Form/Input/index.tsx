/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * The contents of this file are subject to the terms of the Liferay Enterprise
 * Subscription License ("License"). You may not use this file except in
 * compliance with the License. You can obtain a copy of the License by
 * contacting Liferay, Inc. See the License for the specific language governing
 * permissions and limitations under the License, including but not limited to
 * distribution rights of the Software.
 */

import {ClayInput} from '@clayui/form';
import {InputHTMLAttributes} from 'react';

import BaseWrapper from '../Base/BaseWrapper';

type InputTypes = {
	errors?: any;
	id?: string;
	label?: string;
	name: string;
	register?: any;
	required?: boolean;
	type?: string;
} & InputHTMLAttributes<HTMLInputElement>;

const Input: React.FC<InputTypes> = ({
	errors = {},
	label,
	name,
	register = () => {},
	id = name,
	type,
	value,
	required = false,
	onBlur,
	...otherProps
}) => {
	return (
		<>
			<BaseWrapper
				error={errors[name]?.message}
				id={id}
				label={label}
				required={required}
			>
				<ClayInput
					className="rounded-xs"
					component={type === 'textarea' ? 'textarea' : 'input'}
					id={id}
					name={name}
					type={type}
					value={value}
					{...otherProps}
					{...register(name, {onBlur, required})}
				/>
			</BaseWrapper>
		</>
	);
};

export default Input;
