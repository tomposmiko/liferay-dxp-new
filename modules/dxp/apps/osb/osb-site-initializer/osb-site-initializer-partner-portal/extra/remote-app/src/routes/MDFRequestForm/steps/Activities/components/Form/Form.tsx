/* eslint-disable react-hooks/exhaustive-deps */
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * The contents of this file are subject to the terms of the Liferay Enterprise
 * Subscription License ("License"). You may not use this file except in
 * compliance with the License. You can obtain a copy of the License by
 * contacting Liferay, Inc. See the License for the specific language governing
 * permissions and limitations under the License, including but not limited to
 * distribution rights of the Software.
 */

import {useEffect, useMemo} from 'react';

import PRMForm from '../../../../../../common/components/PRMForm';
import PRMFormik from '../../../../../../common/components/PRMFormik';
import {TypeActivityKey} from '../../../../../../common/enums/TypeActivityKey';
import {LiferayPicklistName} from '../../../../../../common/enums/liferayPicklistName';
import {TacticKeys} from '../../../../../../common/enums/mdfRequestTactics';
import LiferayPicklist from '../../../../../../common/interfaces/liferayPicklist';
import MDFRequestActivity from '../../../../../../common/interfaces/mdfRequestActivity';
import getNewActivity from '../../utils/getNewActivity';
import BudgetBreakdownSection from './components/BudgetBreakdownSection';
import ContentMarketingFields from './components/ContentMarketingFields';
import DigitalMarketingFields from './components/DigitalMarketingFields';
import EventFields from './components/EventFields';
import LeadListSection from './components/LeadListSection';
import MiscellaneousMarketingFields from './components/MiscellaneousMarketingFields';
import useDynamicFieldEntries from './hooks/useDynamicFieldEntries';
import useTacticsOptions from './hooks/useTacticsOptions';
import useTypeActivityOptions from './hooks/useTypeActivityOptions';

interface IProps {
	currency: LiferayPicklist;
	currentActivity: MDFRequestActivity;
	currentActivityIndex: number;
	setFieldValue: (
		field: string,
		value: any,
		shouldValidate?: boolean | undefined
	) => void;
}

type TypeActivityComponent = {
	[key in string]?: JSX.Element;
};

const Form = ({
	currency,
	currentActivity,
	currentActivityIndex,
	setFieldValue,
}: IProps) => {
	const {fieldEntries} = useDynamicFieldEntries();

	const handleClearForm = () => {
		setFieldValue(
			`activities[${currentActivityIndex}].activityDescription`,
			getNewActivity(currency).activityDescription
		);
	};

	useEffect(() => {
		const displaySection =
			currentActivity?.typeActivity?.key === TypeActivityKey.EVENT
				? 'true'
				: '';

		setFieldValue(
			`activities[${currentActivityIndex}].activityDescription.leadGenerated`,
			displaySection
		);
	}, [currentActivity.tactic, currentActivity?.typeActivity?.key]);

	const {
		onTypeActivitySelected,
		typeActivitiesOptions,
	} = useTypeActivityOptions(
		fieldEntries[LiferayPicklistName.TYPE_OF_ACTIVITY],
		(selectedTypeActivity) => {
			setFieldValue(
				`activities[${currentActivityIndex}].typeActivity`,
				selectedTypeActivity
			);

			setFieldValue(`activities[${currentActivityIndex}].tactic`, {});
		},
		handleClearForm
	);

	const {onTacticSelected, tacticsOptions} = useTacticsOptions(
		useMemo(
			() =>
				fieldEntries[LiferayPicklistName.TACTIC]?.filter((tactic) =>
					String(tactic.value).includes(
						String(currentActivity.typeActivity?.key)
					)
				),
			[currentActivity.typeActivity?.key, fieldEntries]
		),
		(selectedTactic) =>
			setFieldValue(
				`activities[${currentActivityIndex}].tactic`,
				selectedTactic
			),
		handleClearForm
	);

	const typeActivityComponents: TypeActivityComponent = {
		[TypeActivityKey.DIGITAL_MARKETING]: (
			<DigitalMarketingFields
				currentActivityIndex={currentActivityIndex}
				tactic={currentActivity.tactic?.key as TacticKeys}
			/>
		),
		[TypeActivityKey.CONTENT_MARKETING]: (
			<ContentMarketingFields
				currentActivityIndex={currentActivityIndex}
			/>
		),
		[TypeActivityKey.EVENT]: (
			<EventFields
				currentActivityIndex={currentActivityIndex}
				tactic={currentActivity.tactic?.key as TacticKeys}
			/>
		),
		[TypeActivityKey.MISCELLANEOUS_MARKETING]: (
			<MiscellaneousMarketingFields
				currentActivityIndex={currentActivityIndex}
				tactic={currentActivity.tactic?.key as TacticKeys}
			/>
		),
	};
	useEffect(() => {
		setFieldValue(`activities[${currentActivityIndex}].currency`, currency);
	}, []);

	return (
		<>
			<PRMForm.Section title="Campaign Activity">
				<PRMFormik.Field
					component={PRMForm.InputText}
					label="Activity name"
					name={`activities[${currentActivityIndex}].name`}
					required
				/>

				<PRMForm.Group>
					<PRMFormik.Field
						component={PRMForm.Select}
						label="Type of Activity"
						name={`activities[${currentActivityIndex}].typeActivity`}
						onChange={onTypeActivitySelected}
						options={typeActivitiesOptions}
						required
					/>

					<PRMFormik.Field
						component={PRMForm.Select}
						emptyOptionMessage="Select a Type of Activity"
						label="Tactic"
						name={`activities[${currentActivityIndex}].tactic`}
						onChange={onTacticSelected}
						options={tacticsOptions}
						required
					/>
				</PRMForm.Group>

				{
					typeActivityComponents[
						String(currentActivity.typeActivity?.key) || ''
					]
				}

				<LeadListSection
					currentActivityIndex={currentActivityIndex}
					fieldEntries={fieldEntries}
					selectedTypeActivity={String(
						currentActivity.typeActivity?.key
					)}
				/>

				<PRMForm.Group>
					<PRMFormik.Field
						component={PRMForm.DatePicker}
						label="Start Date"
						name={`activities[${currentActivityIndex}].startDate`}
						required
					/>

					<PRMFormik.Field
						component={PRMForm.DatePicker}
						label="End Date"
						name={`activities[${currentActivityIndex}].endDate`}
						required
					/>
				</PRMForm.Group>
			</PRMForm.Section>
			<PRMFormik.Array
				budgets={currentActivity.budgets}
				component={BudgetBreakdownSection}
				currency={currency}
				currentActivityIndex={currentActivityIndex}
				expenseEntries={
					fieldEntries[LiferayPicklistName.BUDGET_EXPENSES]
				}
				name={`activities[${currentActivityIndex}].budgets`}
				setFieldValue={setFieldValue}
			/>
			<div className="d-none total-mdf-request-amount">
				{`Total MDF Requested Amount: ${currentActivity.mdfRequestAmount}`}
			</div>
		</>
	);
};

export default Form;
