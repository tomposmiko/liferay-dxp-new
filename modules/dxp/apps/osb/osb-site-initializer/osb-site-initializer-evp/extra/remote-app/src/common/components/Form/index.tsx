/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * The contents of this file are subject to the terms of the Liferay Enterprise
 * Subscription License ("License"). You may not use this file except in
 * compliance with the License. You can obtain a copy of the License by
 * contacting Liferay, Inc. See the License for the specific language governing
 * permissions and limitations under the License, including but not limited to
 * distribution rights of the Software.
 */

import Button from './Button/Button';
import Checkbox from './Checkbox';
import DatePicker from './DatePicker';
import Input from './Input';

const Form = () => {};

Form.Button = Button;
Form.Divider = () => <hr />;
Form.Input = Input;
Form.DatePicker = DatePicker;
Form.Checkbox = Checkbox;

export default Form;
