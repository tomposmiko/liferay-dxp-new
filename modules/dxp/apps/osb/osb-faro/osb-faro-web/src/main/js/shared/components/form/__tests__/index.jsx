import Form from '../index';
import React from 'react';
import {render} from '@testing-library/react';

jest.unmock('react-dom');

describe('Form', () => {
	it('should render a Form', () => {
		const {container} = render(<Form />);

		expect(container).toMatchSnapshot();
	});

	it('should render a Form Group', () => {
		const {container} = render(<Form.Group />);

		expect(container).toMatchSnapshot();
	});

	it('should render an autofit Form Group', () => {
		const {container} = render(<Form.Group autoFit />);

		expect(container.querySelector('.form-group-autofit')).toBeTruthy();
	});
});
