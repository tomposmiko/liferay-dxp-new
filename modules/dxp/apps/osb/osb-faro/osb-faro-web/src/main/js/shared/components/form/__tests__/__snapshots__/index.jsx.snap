// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should render a Form 1`] = `<div />`;

exports[`Form should render a Form Group 1`] = `
<div>
  <div
    class="form-group"
  />
</div>
`;
